docs/src/pages/css-in-js/api/api.md

```diff
@@ -120,6 +120,21 @@ import { ServerStyleSheets } from '@material-ui/styles';
 const sheets = new ServerStyleSheets();
 ```
 
+#### Arguments
+
+The `new ServerStyleSheets([options])` instantiation accepts an options object as a first argument.
+
+1. `options` (*Object* [optional]): The options are spread as props to the [`StylesProvider`](#stylesprovider) component.
+
+  **Examples**
+
+  ```js
+  import { ServerStyleSheets } from '@material-ui/styles';
+  import generateClassName from './generateClassName';
+
+  const sheets = new ServerStyleSheets({ generateClassName });
```

Comment:
`this.options` is only referenced once:
https://github.com/mui-org/material-ui/blob/1fa8f9441d1645d4f1c74c8856f8d72bbbeb6653/packages/material-ui-styles/src/ServerStyleSheets/ServerStyleSheets.js#L24

I don't think it makes sense currently to implement this as a class anyway since it has internal dependencies on the call order.

A functional module would make more sense currently and would've prevented this confusion.

/cc @oliviertassinari

Reply:
I haven't cleanup the logic 🙈 . What do you think of removing the second argument from options and spreading `this.options`?  
```diff
--- a/packages/material-ui-styles/src/ServerStyleSheets/ServerStyleSheets.js
+++ b/packages/material-ui-styles/src/ServerStyleSheets/ServerStyleSheets.js
@@ -8,7 +8,7 @@ class ServerStyleSheets {
     this.options = options;
   }

-  collect(children, options) {
+  collect(children) {
     // This is needed in order to deduplicate the injection of CSS in the page.
     const sheetsManager = new Map();
     // This is needed in order to inject the critical CSS.
@@ -21,8 +21,7 @@ class ServerStyleSheets {
         sheetsManager={sheetsManager}
         serverGenerateClassName={generateClassName}
         sheetsRegistry={this.sheetsRegistry}
-        sheetsCache={this.options.sheetsCache}
-        {...options}
+        {...this.options}
       >
         {children}
       </StylesProvider>
```

Reply:
Oh. Yeah that documentation is currently misleading. You should document usage in a single section. It currently looks like all of these methods are somewhat independent when they actually only work when used on the same instance in that order.

`collect` shouldn't have a separate section. It should only be documented in conjunction with `toString` and `getStyleElement`. It also doesn't return `void`.